### DISCUSSION TITLE
How can we change the page and component display names to other languages

### DISCUSSION THREAD

**Discussion by chendwww** [mui/toolpad#2526] [Help and support]
Hi, we're using tooplad for our project, we found toolpad doesn't allow us to change the page name to other language except English

![image](https://github.com/mui/mui-toolpad/assets/25785677/bf0cd8ff-fd16-4481-beb4-6025e9580250)

![image](https://github.com/mui/mui-toolpad/assets/25785677/bdd56156-49d6-4e5f-87c3-39f6e54029bf)

Is possible to change the page and component names to other languages? 

I see both pages and components have an uuid, so why display names have to be English?

**Comment by Janpot** [comment]
At the moment names of Toolpad objects must be valid javascript identifiers because they will create global variables in binding scripts. This is fine as most of these names are hidden from the end-user anyway. Yet pages are an exception as their names show up in the navigation sidebar. I'm opening https://github.com/mui/mui-toolpad/issues/2527 to propose a way for customizing these.

**Reply by JerryWu1234** [reply]
How does that create global variables in binding scripts?

**Reply by Janpot** [reply]
at the moment, pages don't create values in global scope, but other toolpad objects do (e.g. TextField, ...). I'm in favor of keeping the same rules for all these objects as to keep them adressable with valid JS identifiers

**Reply by JerryWu1234** [reply]
Do we create a new property for the page name ?

**Reply by Janpot** [reply]
We can add a `displayName` property in the `./page/myPage.yml` as:

```yml
# ./page/myPage.yml
apiVersion: v1
kind: page
spec:
  id: vo1091s
  displayName: My Page
```

**Reply by JerryWu1234** [reply]
![image](https://github.com/mui/mui-toolpad/assets/27261522/1fff541e-231f-431b-baf2-3e4f379cc72f)
so it also needs to add a new input there to bind `displayName `

**Reply by Janpot** [reply]
We can add it as an option here:

<img width="444" alt="Screenshot 2023-09-19 at 07 16 42" src="https://github.com/mui/mui-toolpad/assets/2109932/9caeb19a-af06-44a0-9393-9a1d94bada2d">

**Reply by JerryWu1234** [reply]
if we would add it as an option, we should give a default value for page was just created.

**Reply by JerryWu1234** [reply]
I suggest we can use the `name` directly to pass to `displayName` when first creating

**Reply by JerryWu1234** [reply]
I will do it in my next plan.
Please let me know if you have any other requirements ..

**Reply by Janpot** [reply]
I'd like the initial page to be as minimal as possible. e.g. It should be possible to just create a `./pages/myPage/page.yml` with content `kind: 'page'` and it should display a default page.

 So I suggest we don't initialize this property and instead fall back to the file name whenever we want to use the display name (`page.displayName || page.name`)
